easybuild/framework/easyblock.py

```diff
@@ -458,68 +458,85 @@ def fetch_patches(self, patch_specs=None, extension=False, checksums=None):
+                    extra_version = extra[1]
 
-                    # make sure we grab *raw* dict of default options for extension,
+                    # make sure we grab *raw* dict of default options for extension/component,
                     # since it may use template values like %(name)s & %(version)s
-                    ext_options = copy.deepcopy(self.cfg.get_ref('exts_default_options'))
+                    extra_options = copy.deepcopy(self.cfg.get_ref(default_options_string))
 
-                    if len(ext) == 3:
-                        if isinstance(ext_options, dict):
-                            ext_options.update(ext[2])
+                    if len(extra) == 3:
+                        if isinstance(extra_options, dict):
+                            extra_options.update(extra[2])
                         else:
-                            raise EasyBuildError("Unexpected type (non-dict) for 3rd element of %s", ext)
-                    elif len(ext) > 3:
-                        raise EasyBuildError('Extension specified in unknown format (list/tuple too long)')
-
-                    ext_src = {
-                        'name': ext_name,
-                        'version': ext_version,
-                        'options': ext_options,
+                            raise EasyBuildError("Unexpected type (non-dict) for 3rd element of %s", extra)
+                    elif len(extra) > 3:
+                        raise EasyBuildError('The %s specified in unknown format (list/tuple too long)', option_string)
+
+                    extra_src = {
+                        'name': extra_name,
+                        'version': extra_version,
+                        'options': extra_options,
                     }
 
                     # construct dictionary with template values;
                     # inherited from parent, except for name/version templates which are specific to this extension
                     template_values = copy.deepcopy(self.cfg.template_values)
-                    template_values.update(template_constant_dict(ext_src))
+                    template_values.update(template_constant_dict(extra_src, skip_lower=False))
 
                     # resolve templates in extension options
-                    ext_options = resolve_template(ext_options, template_values)
+                    extra_options = resolve_template(extra_options, template_values)
 
-                    checksums = ext_options.get('checksums', [])
+                    checksums = extra_options.get('checksums', [])
 
                     # use default template for name of source file if none is specified
                     default_source_tmpl = resolve_template('%(name)s-%(version)s.tar.gz', template_values)
-                    fn = ext_options.get('source_tmpl', default_source_tmpl)
+                    if components:
+                        fn = resolve_template(extra_src['options'].get('sources', default_source_tmpl)[0],
```

Comment:
Problem here is that for components this is a list, there may be more than one source file